frontend/src/component/admin/network/NetworkTrafficUsage/selectable-periods.test.ts

```diff
@@ -0,0 +1,31 @@
+import { generateSelectablePeriodsFromDate } from './selectable-periods';
+
+test('marks months before May 2024 as unselectable', () => {
+    const now = new Date('2025-01-01');
+    const selectablePeriods = generateSelectablePeriodsFromDate(now);
+
+    expect(
+        selectablePeriods.map(({ key, selectable }) => ({ key, selectable })),
+    ).toEqual([
+        { key: '2025-01', selectable: true },
+        { key: '2024-12', selectable: true },
+        { key: '2024-11', selectable: true },
+        { key: '2024-10', selectable: true },
+        { key: '2024-09', selectable: true },
+        { key: '2024-08', selectable: true },
+        { key: '2024-07', selectable: true },
+        { key: '2024-06', selectable: true },
+        { key: '2024-05', selectable: true },
+        { key: '2024-04', selectable: false },
+        { key: '2024-03', selectable: false },
+        { key: '2024-02', selectable: false },
+    ]);
+});
+
+test('generates 12 months, including the current month', () => {
```

Comment:
Did the old implementation also generate 12 months? I think we said 12 + 1 originally

Reply:
Maybe, but in this one we only want twelve. The grid month selector shouldn't show more than 12 months (that'd break the 4 x 3 grid), so that's where this one's capped. Happy to change that later if we get designs for it, though.